### PR TITLE
MOTECH docs

### PR DESCRIPTION
**Pull request by kaapstorm**
##### SUMMARY

This is a draft PR to get feedback on my choices and implementation of:
* using ReStructuredText
* moving technical documentation into docstrings
* preparing MOTECH docs for inclusion into commcare-hq.readthedocs.io

I have not yet tested building these docs.

:tropical_fish: or :classical_building: 

cc @karmuno @avazirna

### CODE REVIEW

**Comment by orangejenny** [comment]
@kaapstorm Would you be open to adding these decisions (specifically RST usage and docstrings) to someplace like CONTRIBUTING.rst and running them by the entire team? I have some opinions on these things, but more than that I'd love us to have guidelines that everyone is bought into.

My two specific thoughts on this PR are
- I appreciate that you reference the dev-facing docs in the readthedocs, since it mitigates the problem of people looking in the wrong places for docs.
- It's unfortunate that GitHub can't handle preview for docstrings the way it does for dedicated markdown and rst files; I use preview quite a bit both in writing docs and in reviewing them.

**Comment by kaapstorm** [comment]
I've tested building. Changing from "Draft" to normal PR.

File: corehq/motech/README.md [hunk 1]
```diff
@@ -107,25 +106,28 @@ with OpenMRS.
...
-CommCare sends data to OpenMRS using its Web Services API.
+MOTECH sends CommCare data to OpenMRS using OpenMRS's Web Services API.
 
 All data sent to OpenMRS relates to what OpenMRS refers to as
-"patients", "visits", "encounters" and "observations". In CommCare these
-correspond to properties of one or a handful of case types, and values
-of form questions.
+"patients", "visits", "encounters" and "observations". Values from
+CommCare can be retrieved from case properties, form questions,
+locations, *inter alia*.
```
Comment (esoergel) [inline corehq/motech/README.md]: > inter alia

nescio hoc verbo

File: corehq/motech/docs/value_source.rst [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    }
+
+Omit ``direction``, or set it to ``null``, for values that should be
+both imported and exported.
+
+
+CaseProperty
+^^^^^^^^^^^^
+
+.. autoclass:: corehq.motech.value_source.CaseProperty
```
Comment (esoergel) [inline corehq/motech/docs/value_source.rst]: You may also try using `automodule` with some [combination of toggles](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) to pull in relevant docs without declaring each class here.

**Review by esoergel — APPROVED** [review]
I didn't read every word here, but I'm very much on board with the restructuring and format changes.  This is fairly similar to what I did with the [`es_query` module](https://commcare-hq.readthedocs.io/es_query.html).

I do think rst is nice to use for anything non-trivial, since it supports directives like inserting documentation from elsewhere and building tables of contents.  In case you're not aware of the tool, I've had success in the past using `pandoc` to convert existing markdown docs to rst:
```
$ pandoc -o FILENAME.rst FILENAME.md
```

File: corehq/motech/README.md (hunk 1 above)
Comment (kaapstorm) [reply corehq/motech/README.md]: lol, sorry, I'll change it to "among others" or "etc."

File: corehq/motech/docs/value_source.rst (hunk 2 above)
Comment (kaapstorm) [reply corehq/motech/docs/value_source.rst]: :+1: Nice.

**Pull request merged**